app/controllers/application_controller.rb

```diff
@@ -1,10 +1,88 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
+    register Sinatra::ActiveRecordExtension
+    set :views, Proc.new { File.join(root, "../views/") }
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
+    configure do
+      enable :sessions
+      set :session_secret, "secret"
+    end
+
+  get '/' do
+    erb :home
+  end
+
+  get '/index' do
+    erb :index
+  end
+
+  get '/tweets/new' do
+      @user = Helpers.current_user(session)
+      if Helpers.is_logged_in?(session) == false
+          redirect '/login'
+      else
+          erb :'tweets/new'
+      end
+  end
+
+  post '/tweets/new' do
+      if params[:content] == ""
+          redirect '/tweets/new'
+      else
+          @tweet = Tweet.new(:content => params[:content], :user_id => session[:user_id])
+          @tweet.save
+          redirect '/tweets'
+      end
+  end
+
+  get '/tweets/:id' do
+    @user = Helpers.current_user(session)
+    if Helpers.is_logged_in?(session) == false
+      redirect '/login'
+    else
+        @tweet = Tweet.find_by(id: params[:id])
+        erb :'tweets/show'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    @user = Helpers.current_user(session)
+    if Helpers.is_logged_in?(session) == false
+      redirect '/login'
+    else
+        @tweet = Tweet.find_by(id: params[:id])
+        if @tweet.user_id == session[:user_id]
+            erb :'tweets/edit'
+        else
+            erb :error
+        end
+    end
+  end
+
+  patch '/tweets/:id' do
+      @tweet = Tweet.find_by(id: params[:id])
+      if params[:content] == ""
```

Comment:
Another good place to just look for the current user's tweets, since we should only be modifying tweets we own:
```
@tweet = current_user.tweets.find_by(id: params[:id])
if @tweet
  #...
```